pkg/target/regolib/src.rego

```diff
@@ -273,10 +273,18 @@ matches_namespaces(match) {
 
 matches_namespaces(match) {
   has_field(match, "namespaces")
+  not is_ns(input.review.kind)
   ns := {n | n = match.namespaces[_]}
   count({input.review.namespace} - ns) == 0
 }
 
+matches_namespaces(match) {
+  has_field(match, "namespaces")
+  is_ns(input.review.kind)
+  ns := {n | n = match.namespaces[_]}
+  count({input.review.object.metadata.name} - ns) == 0
```

Comment:
should we add this logic to `get_ns[ns]`

Reply:
get_ns returns the full namespace object, here we are looking for the namespace name.

There may be some value in keeping them separate as we should have enough info to get the namespace name without requiring caching or sideloading, the full namespace object needs at least one of those two to work.

Reply:
Updated to use the same `get_ns_name[]` rule as #433